cpp/src/lists/contains.cu

```diff
@@ -51,29 +52,74 @@ auto get_pair_iterator(cudf::column_device_view const& d_search_keys)
   return d_search_keys.pair_begin<ElementType, has_nulls>();
 }
 
-template <typename ElementType, bool>
+template <typename ElementType,
+          bool,
+          std::enable_if_t<!cudf::is_fixed_point<ElementType>(), void>* = nullptr>
 auto get_pair_iterator(cudf::scalar const& search_key)
 {
   return cudf::detail::make_pair_iterator<ElementType>(search_key);
 }
 
+template <typename ElementType,
+          bool,
+          std::enable_if_t<cudf::is_fixed_point<ElementType>(), void>* = nullptr>
+auto get_pair_iterator(cudf::scalar const& search_key)
+{
+  return cudf::detail::make_counting_transform_iterator(
+    0, cudf::detail::scalar_representation_pair_accessor<ElementType>{search_key});
+}
+
+/**
+ * @brief Custom equality comparator for lists::contains().
+ *
+ * Special handling for fixed_point types. Ignores scale values and compares
+ * only their representation. (Scales have already been normalized by this point.)
+ * For non-fixed_point types, comparisons are delegated to `cudf::equality_compare`.
+ */
+struct equality_comparator {
```

Comment:
If `pair_rep_iterator` is used that returns an iterator to the `rep`, then there wouldn't be any need for a custom comparator, yes?

Reply:
Hmm... Good question. 

I *thought* this was required because `make_pair_rep_iterator()` holds for the `scalar` argument. The `lhs` of the comparison is an element from a list, which does not benefit from `make_pair_rep_iterator()`.
i.e. `equality_compare(decimal32, int32_t)` wouldn't compile.

`equality_comparator` might be obviated if we had a `rep_pair_begin()` for use with `column_device_view`.

I'll take a crack at that approach. Thanks, this might be cleaner.

Reply:
>> If pair_rep_iterator is used that returns an iterator to the rep...

> `equality_comparator` might be obviated if we had a `rep_pair_begin()` for use with `column_device_view`

As much as I would've liked to remove `equality_comparator`, it looks like this change is not feasible here. For reference:
```c++
d_bools[row_index] =
  thrust::find_if(thrust::seq,
                  list.pair_begin<ElementType>(),
                  list.pair_end<ElementType>(),
                  [search_key] __device__(auto element_and_validity) {
                    return element_and_validity.second &&
                           equality_comparator{}(element_and_validity.first, search_key);
                  }) != list.pair_end<ElementType>();
```
`list.pair_begin()` is being called on a `list_device_view` and not on a `cudf::column_view` or a `scalar`. The iteration is for all elements in a list row. :/

Reply:
I was able to find a way around it:
1. I construct a `transform_iterator` around `list.pair_begin<E>` to extract the `rep()` value in case of decimals.
2. I needed to do the same for `column_view.pair_begin<E>`, to support search-key columns.
With these in place, in case of decimal operands, the representative integral values are compared instead.